<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.7.2" />
<title>ieugwaspy.query API documentation</title>
<meta name="description" content="API query functions" />
<link href='https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.0/normalize.min.css' rel='stylesheet'>
<link href='https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/8.0.0/sanitize.min.css' rel='stylesheet'>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet">
<style>.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{font-weight:bold}#index h4 + ul{margin-bottom:.6em}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>ieugwaspy.query</code></h1>
</header>
<section id="section-intro">
<p>API query functions</p>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">&#34;&#34;&#34;API query functions
&#34;&#34;&#34;
import ieugwaspy.constants as cons
import ieugwaspy.variants as variants
import ieugwaspy.backwards as backwards
import json, requests, os


def api_query(path, query=&#34;&#34;, method=&#34;GET&#34;, access_token=cons.api_token):
    &#34;&#34;&#34;This is a general-purpose function called by other functions to query the API and return the resulting data in JSON format.  

    Parameters:
        path: the path component of the URL
        query: any query to go after the ?query component of the URL
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;  # Still need to handle timeouts
    ntry = 0
    ntries = 3
    if method == &#34;GET&#34;:
        if query != &#34;&#34;:
            url = &#34;{}{}?query={}&amp;access_token={}&#34;.format(
                cons.option[&#34;mrbaseapi&#34;], path, query, access_token
            )
        else:
            url = &#34;{}{}?access_token={}&#34;.format(
                cons.option[&#34;mrbaseapi&#34;], path, access_token
            )
        print(url)
        response = requests.get(url)
    elif method == &#34;POST&#34;:
        url = &#34;{}{}&#34;.format(cons.option[&#34;mrbaseapi&#34;], path)
        response = requests.post(url, data=query)
    else:
        return &#34;Invalid API method&#34;
    data = json.loads(response.text)
    return data


def api_status():
    &#34;&#34;&#34;Check API status

    Returns:
        data: API status 

    &#34;&#34;&#34;
    data = api_query(&#34;status&#34;)
    return data


def gwasinfo(id=&#34;&#34;, access_token=cons.api_token):
    &#34;&#34;&#34;This function will return GWAS study meta-data describing the specific studies selected by id

    Parameters:
        id: ID(s) of specific studies (Python list)
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    id = &#34;,&#34;.join(id)
    if id != &#34;&#34;:
        data = api_query(&#34;gwasinfo/{}&#34;.format(id), access_token=access_token)
    else:
        data = api_query(&#34;gwasinfo/list&#34;, access_token=access_token)
    return data


def associations(
    variantlist,
    id,
    proxies=1,
    r2=0.8,
    align_alleles=1,
    palindromes=1,
    maf_threshold=0.3,
    access_token=cons.api_token,
):
    &#34;&#34;&#34;Retrieve associations for specific SNP/study combinations

    Parameters:
        variantlist: list of variants (Python list)
        id: list of study IDs (Python list)
        proxies: proxies 0 or (default) 1 - indicating whether to look for proxies
        r2: minimum proxy LD rsq value. Default=0.8
        align_alleles: try to align tag alleles to target alleles (if proxies = 1). 1 = yes (default), 0 = no
        palindromes: allow palindromic SNPs (if proxies = 1). 1 = yes (default), 0 = no
        maf_threshold: MAF threshold to try to infer palindromic SNPs. Default = 0.3.
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    id = backwards.legacy_ids(id)
    data = {
        &#34;variant&#34;: variantlist,
        &#34;id&#34;: id,
        &#34;proxies&#34;: proxies,
        &#34;r2&#34;: r2,
        &#34;align_alleles&#34;: align_alleles,
        &#34;palindromes&#34;: palindromes,
        &#34;maf_threshold&#34;: maf_threshold,
        &#34;access_token&#34;: access_token,
    }
    result = api_query(&#34;associations&#34;, query=data, method=&#34;POST&#34;)
    return result


def phewas(variantlist, pval=1e-3, access_token=cons.api_token):
    &#34;&#34;&#34;Perform phenome-wide association analysis (PheWAS) of variant(s) across all traits in the IEU GWAS database

    Parameters:
        variants: list of variants (Python list)
        pval: p-value threshold
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    rsid = &#34;,&#34;.join(variants.variants_to_rsid(variantlist))
    data = api_query(&#34;phewas/{}/{}&#34;.format(rsid, pval), access_token=access_token)
    return data


def tophits(
    id,
    pval=5e-8,
    clump=1,
    r2=0.001,
    kb=10000,
    force_server=False,
    access_token=cons.api_token,
):
    &#34;&#34;&#34;Retrieve association data for specific SNP/study combinations 

    Parameters:
        id: list of study IDs (Python list)
        pval: use this p-value threshold. Default = 5e-8
        clump: whether to clump (1) or not (0). Default = 1
        r2: use this clumping r2 threshold. Default is very strict, 0.001
        kb: use this clumping kb window. Default is very strict, 10000
        force_server: True/False. By default (False) will return preclumped hits. p-value threshold 5e-8, with r2 threshold 0.001 and kb threshold 10000, using only SNPs with MAF &gt; 0.01 in the European samples in 1000 genomes. If force_server = TRUE then will recompute using server side LD reference panel.
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    id = backwards.legacy_ids(id)
    if clump == 1 and r2 == 0.001 and kb == 10000 and pval == 5e-8:
        preclumped = 1
    else:
        preclumped = 0
    if preclumped == 1 and force_server:
        preclumped = 0
    data = {
        &#34;id&#34;: id,
        &#34;pval&#34;: pval,
        &#34;preclumped&#34;: preclumped,
        &#34;clump&#34;: clump,
        &#34;r2&#34;: r2,
        &#34;kb&#34;: kb,
        &#34;access_token&#34;: access_token,
    }
    result = api_query(&#34;tophits&#34;, query=data, method=&#34;POST&#34;)
    return result</code></pre>
</details>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="ieugwaspy.query.api_query"><code class="name flex">
<span>def <span class="ident">api_query</span></span>(<span>path, query='', method='GET', access_token='NULL')</span>
</code></dt>
<dd>
<section class="desc"><p>This is a general-purpose function called by other functions to query the API and return the resulting data in JSON format.
</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>path</code></strong></dt>
<dd>the path component of the URL</dd>
<dt><strong><code>query</code></strong></dt>
<dd>any query to go after the ?query component of the URL</dd>
<dt><strong><code>access_token</code></strong></dt>
<dd>the OAuth access token</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>data</code></strong></dt>
<dd>json object as returned by API</dd>
</dl></section>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def api_query(path, query=&#34;&#34;, method=&#34;GET&#34;, access_token=cons.api_token):
    &#34;&#34;&#34;This is a general-purpose function called by other functions to query the API and return the resulting data in JSON format.  

    Parameters:
        path: the path component of the URL
        query: any query to go after the ?query component of the URL
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;  # Still need to handle timeouts
    ntry = 0
    ntries = 3
    if method == &#34;GET&#34;:
        if query != &#34;&#34;:
            url = &#34;{}{}?query={}&amp;access_token={}&#34;.format(
                cons.option[&#34;mrbaseapi&#34;], path, query, access_token
            )
        else:
            url = &#34;{}{}?access_token={}&#34;.format(
                cons.option[&#34;mrbaseapi&#34;], path, access_token
            )
        print(url)
        response = requests.get(url)
    elif method == &#34;POST&#34;:
        url = &#34;{}{}&#34;.format(cons.option[&#34;mrbaseapi&#34;], path)
        response = requests.post(url, data=query)
    else:
        return &#34;Invalid API method&#34;
    data = json.loads(response.text)
    return data</code></pre>
</details>
</dd>
<dt id="ieugwaspy.query.api_status"><code class="name flex">
<span>def <span class="ident">api_status</span></span>(<span>)</span>
</code></dt>
<dd>
<section class="desc"><p>Check API status</p>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>data</code></strong></dt>
<dd>API status</dd>
</dl></section>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def api_status():
    &#34;&#34;&#34;Check API status

    Returns:
        data: API status 

    &#34;&#34;&#34;
    data = api_query(&#34;status&#34;)
    return data</code></pre>
</details>
</dd>
<dt id="ieugwaspy.query.associations"><code class="name flex">
<span>def <span class="ident">associations</span></span>(<span>variantlist, id, proxies=1, r2=0.8, align_alleles=1, palindromes=1, maf_threshold=0.3, access_token='NULL')</span>
</code></dt>
<dd>
<section class="desc"><p>Retrieve associations for specific SNP/study combinations</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>variantlist</code></strong></dt>
<dd>list of variants (Python list)</dd>
<dt><strong><code>id</code></strong></dt>
<dd>list of study IDs (Python list)</dd>
<dt><strong><code>proxies</code></strong></dt>
<dd>proxies 0 or (default) 1 - indicating whether to look for proxies</dd>
<dt><strong><code>r2</code></strong></dt>
<dd>minimum proxy LD rsq value. Default=0.8</dd>
<dt><strong><code>align_alleles</code></strong></dt>
<dd>try to align tag alleles to target alleles (if proxies = 1). 1 = yes (default), 0 = no</dd>
<dt><strong><code>palindromes</code></strong></dt>
<dd>allow palindromic SNPs (if proxies = 1). 1 = yes (default), 0 = no</dd>
<dt><strong><code>maf_threshold</code></strong></dt>
<dd>MAF threshold to try to infer palindromic SNPs. Default = 0.3.</dd>
<dt><strong><code>access_token</code></strong></dt>
<dd>the OAuth access token</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>data</code></strong></dt>
<dd>json object as returned by API</dd>
</dl></section>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def associations(
    variantlist,
    id,
    proxies=1,
    r2=0.8,
    align_alleles=1,
    palindromes=1,
    maf_threshold=0.3,
    access_token=cons.api_token,
):
    &#34;&#34;&#34;Retrieve associations for specific SNP/study combinations

    Parameters:
        variantlist: list of variants (Python list)
        id: list of study IDs (Python list)
        proxies: proxies 0 or (default) 1 - indicating whether to look for proxies
        r2: minimum proxy LD rsq value. Default=0.8
        align_alleles: try to align tag alleles to target alleles (if proxies = 1). 1 = yes (default), 0 = no
        palindromes: allow palindromic SNPs (if proxies = 1). 1 = yes (default), 0 = no
        maf_threshold: MAF threshold to try to infer palindromic SNPs. Default = 0.3.
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    id = backwards.legacy_ids(id)
    data = {
        &#34;variant&#34;: variantlist,
        &#34;id&#34;: id,
        &#34;proxies&#34;: proxies,
        &#34;r2&#34;: r2,
        &#34;align_alleles&#34;: align_alleles,
        &#34;palindromes&#34;: palindromes,
        &#34;maf_threshold&#34;: maf_threshold,
        &#34;access_token&#34;: access_token,
    }
    result = api_query(&#34;associations&#34;, query=data, method=&#34;POST&#34;)
    return result</code></pre>
</details>
</dd>
<dt id="ieugwaspy.query.gwasinfo"><code class="name flex">
<span>def <span class="ident">gwasinfo</span></span>(<span>id='', access_token='NULL')</span>
</code></dt>
<dd>
<section class="desc"><p>This function will return GWAS study meta-data describing the specific studies selected by id</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>id</code></strong></dt>
<dd>ID(s) of specific studies (Python list)</dd>
<dt><strong><code>access_token</code></strong></dt>
<dd>the OAuth access token</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>data</code></strong></dt>
<dd>json object as returned by API</dd>
</dl></section>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def gwasinfo(id=&#34;&#34;, access_token=cons.api_token):
    &#34;&#34;&#34;This function will return GWAS study meta-data describing the specific studies selected by id

    Parameters:
        id: ID(s) of specific studies (Python list)
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    id = &#34;,&#34;.join(id)
    if id != &#34;&#34;:
        data = api_query(&#34;gwasinfo/{}&#34;.format(id), access_token=access_token)
    else:
        data = api_query(&#34;gwasinfo/list&#34;, access_token=access_token)
    return data</code></pre>
</details>
</dd>
<dt id="ieugwaspy.query.phewas"><code class="name flex">
<span>def <span class="ident">phewas</span></span>(<span>variantlist, pval=0.001, access_token='NULL')</span>
</code></dt>
<dd>
<section class="desc"><p>Perform phenome-wide association analysis (PheWAS) of variant(s) across all traits in the IEU GWAS database</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>variants</code></strong></dt>
<dd>list of variants (Python list)</dd>
<dt><strong><code>pval</code></strong></dt>
<dd>p-value threshold</dd>
<dt><strong><code>access_token</code></strong></dt>
<dd>the OAuth access token</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>data</code></strong></dt>
<dd>json object as returned by API</dd>
</dl></section>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def phewas(variantlist, pval=1e-3, access_token=cons.api_token):
    &#34;&#34;&#34;Perform phenome-wide association analysis (PheWAS) of variant(s) across all traits in the IEU GWAS database

    Parameters:
        variants: list of variants (Python list)
        pval: p-value threshold
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    rsid = &#34;,&#34;.join(variants.variants_to_rsid(variantlist))
    data = api_query(&#34;phewas/{}/{}&#34;.format(rsid, pval), access_token=access_token)
    return data</code></pre>
</details>
</dd>
<dt id="ieugwaspy.query.tophits"><code class="name flex">
<span>def <span class="ident">tophits</span></span>(<span>id, pval=5e-08, clump=1, r2=0.001, kb=10000, force_server=False, access_token='NULL')</span>
</code></dt>
<dd>
<section class="desc"><p>Retrieve association data for specific SNP/study combinations </p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>id</code></strong></dt>
<dd>list of study IDs (Python list)</dd>
<dt><strong><code>pval</code></strong></dt>
<dd>use this p-value threshold. Default = 5e-8</dd>
<dt><strong><code>clump</code></strong></dt>
<dd>whether to clump (1) or not (0). Default = 1</dd>
<dt><strong><code>r2</code></strong></dt>
<dd>use this clumping r2 threshold. Default is very strict, 0.001</dd>
<dt><strong><code>kb</code></strong></dt>
<dd>use this clumping kb window. Default is very strict, 10000</dd>
<dt><strong><code>force_server</code></strong></dt>
<dd>True/False. By default (False) will return preclumped hits. p-value threshold 5e-8, with r2 threshold 0.001 and kb threshold 10000, using only SNPs with MAF &gt; 0.01 in the European samples in 1000 genomes. If force_server = TRUE then will recompute using server side LD reference panel.</dd>
<dt><strong><code>access_token</code></strong></dt>
<dd>the OAuth access token</dd>
</dl>
<h2 id="returns">Returns</h2>
<dl>
<dt><strong><code>data</code></strong></dt>
<dd>json object as returned by API</dd>
</dl></section>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">def tophits(
    id,
    pval=5e-8,
    clump=1,
    r2=0.001,
    kb=10000,
    force_server=False,
    access_token=cons.api_token,
):
    &#34;&#34;&#34;Retrieve association data for specific SNP/study combinations 

    Parameters:
        id: list of study IDs (Python list)
        pval: use this p-value threshold. Default = 5e-8
        clump: whether to clump (1) or not (0). Default = 1
        r2: use this clumping r2 threshold. Default is very strict, 0.001
        kb: use this clumping kb window. Default is very strict, 10000
        force_server: True/False. By default (False) will return preclumped hits. p-value threshold 5e-8, with r2 threshold 0.001 and kb threshold 10000, using only SNPs with MAF &gt; 0.01 in the European samples in 1000 genomes. If force_server = TRUE then will recompute using server side LD reference panel.
        access_token: the OAuth access token

    Returns:
        data: json object as returned by API

    &#34;&#34;&#34;
    id = backwards.legacy_ids(id)
    if clump == 1 and r2 == 0.001 and kb == 10000 and pval == 5e-8:
        preclumped = 1
    else:
        preclumped = 0
    if preclumped == 1 and force_server:
        preclumped = 0
    data = {
        &#34;id&#34;: id,
        &#34;pval&#34;: pval,
        &#34;preclumped&#34;: preclumped,
        &#34;clump&#34;: clump,
        &#34;r2&#34;: r2,
        &#34;kb&#34;: kb,
        &#34;access_token&#34;: access_token,
    }
    result = api_query(&#34;tophits&#34;, query=data, method=&#34;POST&#34;)
    return result</code></pre>
</details>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="ieugwaspy" href="index.html">ieugwaspy</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="two-column">
<li><code><a title="ieugwaspy.query.api_query" href="#ieugwaspy.query.api_query">api_query</a></code></li>
<li><code><a title="ieugwaspy.query.api_status" href="#ieugwaspy.query.api_status">api_status</a></code></li>
<li><code><a title="ieugwaspy.query.associations" href="#ieugwaspy.query.associations">associations</a></code></li>
<li><code><a title="ieugwaspy.query.gwasinfo" href="#ieugwaspy.query.gwasinfo">gwasinfo</a></code></li>
<li><code><a title="ieugwaspy.query.phewas" href="#ieugwaspy.query.phewas">phewas</a></code></li>
<li><code><a title="ieugwaspy.query.tophits" href="#ieugwaspy.query.tophits">tophits</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc"><cite>pdoc</cite> 0.7.2</a>.</p>
</footer>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad()</script>
</body>
</html>